### PR TITLE
docs: improve std::fs::read doc

### LINKED ISSUE #114432: Confusing sentence about errors in `fs::read`
### Location

`fs::read`'s errors section

### Summary

It holds this hard to follow sentence:

"It will also return an error if it encounters while reading an error of a kind other than [`io::ErrorKind::Interrupted`]."
[fs::read](https://doc.rust-lang.org/std/fs/fn.read.html)

### PR DESCRIPTION
**Pull request by SteveLauC**
#### What does this PR do

1. Rephrase a confusing sentence in the document of `std::fs::read()`

-----

Closes #114432 

cc @Dexus0 @saethlin

### CODE REVIEW

**Comment by gurry** [comment]
Perhaps we should do more here and also explain what is so specify about  `io::ErrorKind::Interrupted`.  What will happen if  `io::ErrorKind::Interrupted` is as encountered as opposed to errors "other than  `io::ErrorKind::Interrupted`"? Why do we even bring this error kind up here?

**Comment by SteveLauC** [comment]
> What is so special about io::ErrorKind::Interrupted? Why do we even bring up this error kind here?

Agree, IMO, any error that could be constructed from `Error::from_raw_os_error(an_errno_that_could_be_returned_from_the_read_syscall)` can potentially happen, `Interrupted(EINTR)` is just one of them

**Comment by gurry** [comment]
I see. If that is the case, why do we say it can throw errors other than `Interrupted`? Why not just say it can throw any error that could be constructed from `Error::from_raw_os_error(an_errno_that_could_be_returned_from_the_read_syscall)` (of course word that in a more user-friendly way).

File: library/std/src/fs.rs [hunk 1]
```diff
@@ -233,8 +233,8 @@ pub struct DirBuilder {
 /// This function will return an error if `path` does not already exist.
 /// Other errors may also be returned according to [`OpenOptions::open`].
 ///
-/// It will also return an error if it encounters while reading an error
-/// of a kind other than [`io::ErrorKind::Interrupted`].
+/// Also, while reading, an error of a kind other than [`io::ErrorKind::Interrupted`]
+/// can be returned if it is encountered.
```
Comment (Mark-Simulacrum) [inline library/std/src/fs.rs:237]:
```suggestion
/// Reading from the file handles [`io::ErrorKind::Interrupted`] with automatic retries. See [`io::Read`] documentation for details.
```

https://doc.rust-lang.org/nightly/std/io/trait.Read.html#errors-1 has some further discussion of this -- just a suggestion on possible text here, I think this reads better than both master and the PR proposal. But happy to merge some other form.

Let's also update the similar text on line ~274.

Comment (SteveLauC) [reply library/std/src/fs.rs]: Done:)

**Review by Dexus0 — CHANGES_REQUESTED** [review]
"While reading from the file, handles [`io::ErrorKind::Interrupted`] with retries"

Would make more sense to me.

File: library/std/src/fs.rs [hunk 2]
```diff
@@ -233,8 +233,8 @@ pub struct DirBuilder {
 /// This function will return an error if `path` does not already exist.
 /// Other errors may also be returned according to [`OpenOptions::open`].
 ///
-/// It will also return an error if it encounters while reading an error
-/// of a kind other than [`io::ErrorKind::Interrupted`].
+/// Reading from the file handles [`io::ErrorKind::Interrupted`] with automatic
```
[the remark this answers is not in the document]
Comment (Dexus0) [reply library/std/src/fs.rs:236]: This makes more sense to me.

**Review by Dexus0 — CHANGES_REQUESTED** [review]

File: library/std/src/fs.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (Dexus0) [reply library/std/src/fs.rs:236]: My suggestion seems a wee bit confusing without the next line for context...

[the remark this answers is not in the document]
Comment (SteveLauC) [reply library/std/src/fs.rs:236]: What about adding a subject to the sentence:

While reading from the file, **this function** handles [`io::ErrorKind::Interrupted`] with automatic retries. See [`io::Read`] documentation for details.

[the remark this answers is not in the document]
Comment (Dexus0) [reply library/std/src/fs.rs:236]: Seems good to me.

**Comment by SteveLauC** [comment]
I just pushed the new code comment, ping me if you need me to squash the commits

**Review by Dexus0 — APPROVED** [review]

**Comment by Mark-Simulacrum** [comment]
r=me with commits squashed

**Comment by SteveLauC** [comment]
> > r=me with commits squashed
> 
> @SteveLauC

Does this instruction mean that I need to squash my commits? I thought it can be done automatically

**Comment by saethlin** [comment]
You need to squash your commits. GitHub can do it automatically if you merge via GitHub but the Rust organization does not use GitHub to merge commits. Rust uses an implementation of the same concept behind GitHub merge queues- you only merge if the merged state would pass the tests. Which means you must test merges one-at-a-time. Here's the queue for this repo: https://bors.rust-lang.org/queue/rust

People say "r=me with..." to indicate that anyone else with review permissions can send this off to the merge queue when the thing is done.

**Comment by SteveLauC** [comment]
> You need to squash your commits. GitHub can do it automatically if you merge via GitHub but the Rust organization does not use GitHub to merge commits. Rust uses an implementation of the same concept behind GitHub merge queues- you only merge if the merged state would pass the tests. Which means you must test merges one-at-a-time. Here's the queue for this repo: https://bors.rust-lang.org/queue/rust
> 
> People say "r=me with..." to indicate that anyone else with review permissions can send this off to the merge queue when the thing is done.

Commits squashed,  thanks for you detaild explanation:)
